### DISCUSSION TITLE
Security per single user in semaphore

### DISCUSSION THREAD

**Discussion by mwxp06** [semaphoreui/semaphore#791] [Ideas]
Hallo,
I have a lot of user in semaphore and my problem is, that every user have their own private key and public key on remote servers (where they want to run playbook).

Is there any option to have every user his private credential (not visible to each other) which will be used in playbook run?
Is there any possibility to connect semaphore with Vault by HashiCorp where everyone can save his credential.. 
For example:

- user save his token (or key or pass) for Vault in private credential in semaphore 
- this token will be used to take secured credential from Vault when running job

I think that this possibility or some security like this is not available yet.

**Comment by fiftin** [comment]
Currently no, but I think it is good idea.
